### ISSUE TITLE
Conduits became nearly invisible and couldn’t be broken

### ISSUE THREAD

**Issue by PapiDimmi** [SleepyTrousers/EnderIO#4153]
![2017-04-11_20 55 15](https://cloud.githubusercontent.com/assets/25916106/24925656/326a3a32-1ef9-11e7-85ae-323d70554924.png)

The conduit connector is nearly invisible, with some lines in its place.

I cannot break it, be it with my hand, a tool, or the yeta wrench (shift-right-clicking). I’ve exited and relaunched my game, but that didn’t work. I have no idea what to do.

The energy conduits currently are not sending power to my powered spawners, and this might have something to do with that; or it might just be a visual bug. Either way, this is a pretty infuriating bug, especially if that’s what’s causing my power problem, which just started happening a few minutes ago, by the way—right before I discovered this invisible conduit bug.

I searched and found out that this _has_ been reported before (#3910), but the issue was closed and labelled as “Code Complete,” which, I assume, means that the bug was fixed, which it has not been.
****
**Minecraft version**: 1.10.2
**Minecraft Forge version**: 12.18.3.2281
**Ender IO version**: 3.1.180
**EnderCore version**: 0.4.1.65-beta

**Comment by ktole1999** [comment]
i am guessing it will break in creative?

**Comment by PapiDimmi** [comment]
@ktole1999 I switched to creative mode and broke the conduit. It worked, but I can’t place a block there. I tried to use /setblock and place air in the conduit’s place, but it gave me the error “The block cannot be placed.”

Help? This is super annoying!

**Comment by PapiDimmi** [comment]
I fixed it. What I did was use /setblock to set the conduit connector to stone (rather than air). Then I just broke the stone block and replaced the conduits.

I do not know how to reproduce this bug.

**Comment by ktole1999** [comment]
what did info say about block when you hovered?

**Comment by PapiDimmi** [comment]
What info?

**Comment by ktole1999** [comment]
when you moused over the invisible conduit what did waila/wawla/top info say

**Comment by PapiDimmi** [comment]
[Here is a GIF of the bug](https://gfycat.com/AnotherAthleticBunny).

**Comment by PapiDimmi** [comment]
[Here is a GIF of the bug](https://gfycat.com/AnotherAthleticBunny).

By the way, this bug was what caused my powered spawners to not get any power. It also prevented my liquid XP from getting transferred to my base.

**Comment by PapiDimmi** [comment]
This happened again but in a different place.

This time, I could break the broken conduit in survival mode.

This is a pretty annoying glitch, as it actually breaks the conduits in that they can’t transfer energy and fluids when they’re in this broken state.

**Comment by HenryLoenwind** [comment]
And for us it's pretty annoying that you want support from us but refuse to cooperate in the slightest. That reporting template is there for a reason.

**Comment by PapiDimmi** [comment]
@HenryLoenwind I included all the information that was needed. The versions, screenshots, description of the bug, and even a GIF. I reposted this issue with the template (#4162), and the only thing that’s different is irrelevant information.

**Comment by PapiDimmi** [comment]
@HenryLoenwind I included all the information that was needed. The versions, screenshots, description of the bug, and even a GIF. I reposted this issue with the template (#4162), and the only thing that’s different is irrelevant information. Closing an issue just because I didn’t use the template is silly.

**Comment by MatthiasMann** [comment]
@PapiDimmi you have repeatedly shown to ignore requests for detailed information, screenshots and other things from several mod authors. As well as wasting time by continuing with pointless discussions or misusing emoji.

And as a general note - the mod authors have the final say of how their mods will behave or not - if you disagree with that then stop using that mod.

If you continue this behavior of yours it might end up with people just closing issues you create without even answering you.

**Comment by PapiDimmi** [comment]
>you have repeatedly shown to ignore requests for detailed information, screenshots and other things from several mod authors

Incorrect. I ignored @TehNut’s command to use a template when reporting bugs, which is highly unnecessary. Why? Well, I included all the necessary information: screenshots, descriptions, and versions. What’s the point of including a “steps to reproduce,” “log,” “describe this issue,” _etc._ section when they’re just going to contain one sentence saying something like “I don’t know how to reproduce this,” “I have no log,” “I’ve already described it,” _etc._? All that’ll do is waste not only my time but also the people reading the issue’s time.

>As well as wasting time by continuing with pointless discussions

If I want to discuss with a mod author, why can’t I do that? If the mod author believes that it’s so pointless, he or she can just start ignoring the discussion or locking the issue.

>misusing emoji.

**Comment by PapiDimmi** [comment]
>you have repeatedly shown to ignore requests for detailed information, screenshots and other things from several mod authors

Incorrect. I ignored @TehNut’s command to use a template when reporting bugs, which is highly unnecessary. Why? Well, I included all the necessary information: screenshots, descriptions, and versions. What’s the point of including a “steps to reproduce,” “log,” “describe this issue,” _etc._ section when they’re just going to contain one sentence saying something like “I don’t know how to reproduce this,” “I have no log,” “I’ve already described it,” _etc._? All that’ll do is waste not only my time but also the people reading the issue’s time.

>As well as wasting time by continuing with pointless discussions

If I want to discuss with a mod author, why can’t I do that? If the mod author believes that it’s so pointless, he or she can just start ignoring the discussion or locking the issue.

>misusing emoji.

Emoji reactions are a feature on GitHub, and I’m using that feature to show my reaction to a post. Please explain how I’m “misusing” this feature. Misusing it would be to react with six different emoji on each comment or something—and even then, it wouldn’t be that big of a deal.

If I downvote or upvote comments on Reddit based on my reaction to the comment, is that misusing the vote feature? If I “like” YouTube videos that I like and “dislike” YouTube videos that I dislike, is that misusing this feature?

>And as a general note - the mod authors have the final say of how their mods will behave or not - if you disagree with that then stop using that mod.

I agree. I should be allowed to ask questions, report bugs, and discuss with mod authors, though.

>If you continue this behavior of yours it might end up with people just closing issues you create without even answering you.

I report bugs and suggest features to mod authors because I want to improve the mod (and in some cases, I am requesting help). Closing all my issues because of a personal issue, _e.g._ disliking me, or because I didn’t use a pointless template, is very, very silly.

**Comment by PapiDimmi** [comment]
x

**Comment by PapiDimmi** [comment]
I don’t get why my new issue was closed.

Look: I love Ender IO, and I love its developers. I don’t have anything against any of you, and I don’t mean to offend any of you in any way. All I want is to get discuss with you and get answers. I want to know what “emoji misuse” means, and why I need to use the report template. I want to discuss and get answers, not argue.

My new issue was closed and locked. Why? I used the template that you told me to use.

**Comment by HenryLoenwind** [comment]
no log, no help.

darn, I don't even know if you're on a server or in single player. Just in case, I need the server log.

**Comment by PapiDimmi** [comment]
I don’t think anything of importance will be recorded in the log, but I’ll be sure to save the log the next time this bug happens.

The thing is, it could happen at any time without me noticing it, so finding the right log would be difficult.

@HenryLoenwind Could you please unlock #4162 and re-open it once I have a log, or do I have to repost the issue again?

**Comment by MatthiasMann** [comment]
@PapiDimmi that's exactly the issue - you make lots of assumptions about what information is in the logs and about which information a mod author need to diagnose the issues. Then you get offended when the authors asks for these things and put a negative emoji on the request for the REQUIRED additional information.

This is what I call emoji abuse and annoying anyone trying to help - so you really should change your behavior.

**Comment by PapiDimmi** [comment]
>that's exactly the issue - you make lots of assumptions about what information is in the logs 

I made _one_ assumption, not “lots of assumptions.”

>Then you get offended when the authors asks for these things

I didn’t get offended. Why are you making lots of assumptions?

>This is what I call emoji abuse and annoying anyone trying to help

I use emoji to react to posts. It’s astonishing how a small Unicode smiley can annoy a person so much.

**Comment by calloatti** [comment]
@PapiDimmi You need to polish your skills regarding interacting with people. You come across as rude and  as an idiot. Sorry if this offends you, just trying to help you out.

**Comment by PapiDimmi** [comment]
Take a guess whether or not calling me an idiot offends me.

If you think it does, don’t say it. Thanks.

**Comment by PapiDimmi** [comment]
Take a guess whether or not calling me an idiot offends me.

If you think it does, refrain from writing it.

**Comment by PapiDimmi** [comment]
Take a guess whether or not calling me an idiot offends me.

If you think it does, please refrain from writing it.

I haven’t tried to offend anyone. I haven’t insulted anyone. I’ve merely discussed with some people who believe some nonsensical things, and I want answers so that I can know why that is. It that makes me an idiot who doesn’t know how to interact wiht people, then so be it.

I’m one step above you, as I have not called anyone an idiot. That’s certainly not an ideal way to interact with others—by calling them an idiot.

**Comment by HenryLoenwind** [comment]
I'd recommend you take a nice long vacation in some ancient monastery and meditate over the difference between "you come across as an idiot" and "you are an idiot".

**Comment by TehNut** [comment]
I wasn't going to come here, but I'm sick of this shit. 

If you want to continue contributing to the development of mods, you need to suck it up and get over your problems with how we run *our* projects. You have absolutely no right to call how I want issues reported "imbecilic". 

Spamming issues with completely irrelevant information is completely unneeded. You claim to be the bigger person, yet you haven't shown that at all.

Continue down the road you're on and I will ban you from every repo I have access to. I have talked about you with countless devs and *none* of them like you or how you handle anything.

So just fucking stop.

**Comment by MatthiasMann** [comment]
@PapiDimmi the reason that we use a template is so that all required data is provided and in a place where we (the mod developers) can quickly find it without reading a wall of text.

It's like your tax form or hundreds of others used to optimize the way people deal with large amounts of data.

So if you ignore the template (or other forms) you not only make the live harder for the people on the other side but you also show that you do not value their time at all.

**Comment by HenryLoenwind** [comment]
Did you know that this issue can be caused by older Mekanism versions? If I had a modlist, I would point that out. It could also be caused by an exception when constructing the network packet on the server, which would be visible in the logfile. Or it could be caused by a new a never before seen bug in Minecraft, Forge, Ender IO or each and every other mod you have installed.

You provided the information "it happens here". All **I** can do with that is saying "I see it happens for you". Nice. Next ticket.

**Comment by PapiDimmi** [comment]
x

**Comment by PapiDimmi** [comment]
@TehNut

>Spamming issues with completely irrelevant information is completely unneeded. 

Spamming means to either repeatedly and excessively post messages _or_ post messages that are unrelated. I have done neither.

>You claim to be the bigger person, yet you haven't shown that at all.

**Comment by PapiDimmi** [comment]
@TehNut

>Spamming issues with completely irrelevant information is completely unneeded. 

Spamming means to either repeatedly and excessively post messages _or_ post messages that are unrelated. I have done neither.

>You claim to be the bigger person, yet you haven't shown that at all.

I never claimed to be the bigger person.

>Continue down the road you're on and I will ban you from every repo I have access to.

All I want is to help the development of the mods that I use by reporting bugs and suggesting features, and sometimes, I try to help others. I think it’s a bit immature to ban me from every single repository you can ban me from, even the ones I’ve never even visited, just because you dislike me.

I do not want to offend anyone. I haven’t insulted anyone like others have insulted me. I just want answers. If you want me to use a pointless template, fine. At least explain what the point of the template is—when I’ve already provided all the necessary information.

At this point, I don’t even care. I _have_ used the template, and I will use it in future issues, since it’s, apparently, so important to you.

>So just fucking stop.

That’s rude. It’s certainly not a way a developer should treat anyone who uses your mods.

@MatthiasMann

>the reason that we use a template is so that all required data is provided and in a place where we (the mod developers) can quickly find it without reading a wall of text.

I use sections, too, and my issues are never a “wall of text.” My issues would be way longer if I were to use a template with several sections that contain information that’s totally irrelevant to the issue; I would have to answer questions that I shouldn’t have to answer in my issue.

>So if you ignore the template (or other forms) you not only make the live harder for the people on the other side but you also show that you do not value their time at all.

I stopped ignoring the template. Like I said, since it’s, apparently, so important, I’m using it now.

@HenryLoenwind

>Did you know that this issue can be caused by older Mekanism versions? If I had a modlist, I would point that out. It could also be caused by an exception when constructing the network packet on the server, which would be visible in the logfile.

Ask me for a mod list, and I’ll give you a mod list. Ask me for a log, and I’ll give you a log.

The thing is that it’s hard to find the right log when this issue can happen at any time without me noticing.

Like I’ve said before, however, I’ll provide my most recent log the next time this happens.

I’ve asked you before, but you didn’t seem to answer me, so I’ll ask you again: Can you unlock the issue #4162, in which I used the template you so desperately want me to use, and then I will provide a log file whenever this issue happens again (up to now, it’s happened twice to me), or do I have to repost the issue for the third time when the bug occurs again?

**Comment by PapiDimmi** [comment]
@TehNut

>Spamming issues with completely irrelevant information is completely unneeded. 

Spamming means to either repeatedly and excessively post messages _or_ post messages that are unrelated. I have done neither.

>You claim to be the bigger person, yet you haven't shown that at all.

I never claimed to be the bigger person.

>Continue down the road you're on and I will ban you from every repo I have access to.

All I want is to help the development of the mods that I use by reporting bugs and suggesting features, and sometimes, I try to help others. I think it’s a bit immature to ban me from every single repository you can ban me from, even the ones I’ve never even visited, just because you dislike me.

I do not want to offend anyone. I haven’t insulted anyone like others have insulted me. I just want answers. If you want me to use a pointless template, fine. At least explain what the point of the template is—when I’ve already provided all the necessary information.

At this point, I don’t even care. I _have_ used the template, and I will use it in future issues, since it’s, apparently, so important to you.

>So just fucking stop.

That’s quite rude. It’s certainly not a way a developer should treat anyone who uses your mods.

@MatthiasMann

>the reason that we use a template is so that all required data is provided and in a place where we (the mod developers) can quickly find it without reading a wall of text.

I use sections, too, and my issues are never a “wall of text.” My issues would be way longer if I were to use a template with several sections that contain information that’s totally irrelevant to the issue; I would have to answer questions that I shouldn’t have to answer in my issue.

>So if you ignore the template (or other forms) you not only make the live harder for the people on the other side but you also show that you do not value their time at all.

I stopped ignoring the template. Like I said, since it’s, apparently, so important, I’m using it now.

@HenryLoenwind

>Did you know that this issue can be caused by older Mekanism versions? If I had a modlist, I would point that out. It could also be caused by an exception when constructing the network packet on the server, which would be visible in the logfile.

Ask me for a mod list, and I’ll give you a mod list. Ask me for a log, and I’ll give you a log.

The thing is that it’s hard to find the right log when this issue can happen at any time without me noticing.

Like I’ve said before, however, I’ll provide my most recent log the next time this happens.

I’ve asked you before, but you didn’t seem to answer me, so I’ll ask you again: Can you unlock the issue #4162, in which I used the template you so desperately want me to use, and then I will provide a log file whenever this issue happens again (up to now, it’s happened twice to me), or do I have to repost the issue for the third time when the bug occurs again?

**Comment by legodude17** [comment]
@PapiDimmi 

> I never claimed to be the bigger person.

You also said this:

> I’m one step above you

> I use sections, too, and my issues are never a “wall of text.”

This one is

Also, you missed several sections from the template, including the steps, what your expected to happen, and your log. The issue template exists so that people know what information is helpful to the maintainers, and they don't have to do through a ton of issues just asking `Where is your log?`, `What steps did you do to reproduce this?`, and so on. If you know enough to know exactly what will help the author fix your issue, then how about you try contributing some code or docs.

**Comment by PapiDimmi** [comment]
>>>I never claimed to be the bigger person.

>>You also said this:

>I’m one step above you

Oh yeah, you’re right. I’m in no way better than anyone else here, but at least I’m not calling people idiots.

>>I use sections, too, and my issues are never a “wall of text.”

>This one is

That’s a _comment_, not an _issue_. Comments don’t have templates, either.

>Also, you missed several sections from the template, including the steps

I don’t know what the steps to reproduce are. What’s the point of that section if all it contains is “I dunno how to reproduce”?

>what your expected to happen

**Comment by PapiDimmi** [comment]
>>>I never claimed to be the bigger person.

>>You also said this:

>I’m one step above you

Oh yeah, you’re right. I’m in no way better than anyone else here, but at least I’m not calling people idiots.

>>I use sections, too, and my issues are never a “wall of text.”

>This one is

That’s a _comment_, not an _issue_. Comments don’t have templates, either.

>Also, you missed several sections from the template, including the steps

I don’t know what the steps to reproduce are. What’s the point of that section if all it contains is “I dunno how to reproduce”?

>what your expected to happen

That’s quite obvious: I expected this bug _not_ to happen. I expected the conduits to work as they normally do. Why would I need to explain that?

>and your log

Like I’ve said multiple times, it’s hard to find the right log. I’ll post a new issue with my most recent log whenever this bug happens again—if it _does_—but I’m not sure it’ll contain any useful information.

**Comment by legodude17** [comment]
@PapiDimmi 

> I don’t know what the steps to reproduce are.

Then do some testing on your own. It is better for everyone to troubleshoot their own issue then for the mod author to be expected to troubleshoot all the issues. Try to remember what you were working on and then do that again. 😃 

> That’s a comment, not an issue.

I meant the issue at the top of this long thread. Sorry for the misunderstanding!

**Comment by MatthiasMann** [comment]
@PapiDimmi regarding "it’s hard to find the right log": once you discover something is wired in your MC world (eg not a crash) then the right log is of course the current one from the world you play. I don't get what's so hard to find these. And the template explicitly asks for that log: "Your most recent log file where the issue was present".

If you would have followed the template when you created the issue the first time - you would have had the log needed. But because you did not include it the log is now lost - which makes debugging this issue much harder.

And even if you are 1000% sure that a log does not contain any information what so ever - that is your opinion. So just assume that the mod developers will find more information in such a log as you would.

**Comment by PapiDimmi** [comment]
@legodude17

>>I don’t know what the steps to reproduce are.

>Then do some testing on your own. It is better for everyone to troubleshoot their own issue then for the mod author to be expected to troubleshoot all the issues. Try to remember what you were working on and then do that again. 😃

The thing is: I have absolutely no idea where to start. This bug seems to occur completely randomly. It would be nearly impossible to figure out what causes it. It might as well just happen randomly.

>>That’s a comment, not an issue.

>I meant the issue at the top of this long thread. Sorry for the misunderstanding!

Oh, I see. No worries!

Anyway, I don’t think that’s a wall of text. It’s much shorter than a long template with irrelevant sections.

But that doesn’t really matter by now, as I’ve said that I’m using the template for all my future issues.

@MatthiasMann

>regarding "it’s hard to find the right log": once you discover something is wired in your MC world (eg not a crash) then the right log is of course the current one from the world you play. I don't get what's so hard to find these. And the template explicitly asks for that log: "Your most recent log file where the issue was present".

Again, I’m not 100% sure that the log will contain any useful information, but again, I promise that the next time this bug occurs to me—if it _does_—I _will_ include my most recent log. I apologize for omitting it from my prior issues.

>If you would have followed the template when you created the issue the first time - you would have had the log needed. But because you did not include it the log is now lost - which makes debugging this issue much harder.

I agree. I always removed all the template text from my issues before starting to type. I do agree that I should’ve included a log, and I’m sorry that I forgot to do so.

>And even if you are 1000% sure that a log does not contain any information what so ever - that is your opinion. So just assume that the mod developers will find more information in such a log as you would.

Yes, I agree. Even if a log doesn’t seem important to me, I’m no mod developer, so chances are that a log could be useful despite my beliefs. I’ll include a log next time.

**Comment by legodude17** [comment]
@PapiDimmi 

> The thing is: I have absolutely no idea where to start.

Try doing whatever you did right before it occured. Were you placing the conduit? In what way? Did you move?

**Comment by PapiDimmi** [comment]
I don’t know _exactly_ when this happens. This could happen hours before I notice.

**Comment by TomyLobo** [comment]
@PapiDimmi 
Probabilistic bugs are a pain to find.
*If*, as you claim, there is no useful information in the logs, you're going to have to wait until someone has an epiphany or a rewrite accidentally fixes the issue.

So, you better wish there was useful information in the logs or that bug is not likely to go away any time soon.

**Comment by PapiDimmi** [comment]
I didn’t claim that there’s no useful information in the logs. I said that I _think_ that there might be no information of significance regarding this bug. But, like I’ve said, I’ll provide a log the next time this bug occurs.

**Comment by Malarix** [comment]
Hi, I came across this issue while googling about this problem. I know this pretty old at this point, but I'm having a very, very similar problem and the issue isn't closed yet, so here's all the info I have that I can give about it (now with template!)

#### Issue Description:

Invisible conduits that no longer make a working connection with surrounding conduits appear on server restart, replacing previously functioning conduits.

#### What happens:

Conduits turn invisible with an outline of the middle part still visible, with other attached conduits appearing to still be connected to it, but without an actual connection being made between them.  The invisible conduits can be either broken or replaced by placing a conduit where they appear to be, but in either case the broken conduit is not dropped as an item or put back in the player's inventory.  This is happening on a multiplayer server and seems to happen on server restart, with conduits being fine before restarting then appearing this way on logging back in.

#### What you expected to happen:

Conduits should remain connected to other conduits, or at least become disconnected properly without still making a psuedo-connection with other conduits.  They should also drop their item when broken or replaced, not disappear entirely. 

#### Steps to reproduce:

1. Place down a long line of conduits on a multiplayer world in a chunkloaded area
2. Restart server
3. Repeat step 2 until conduits turn turn invisible and no longer function.  This has happened randomly for me, and only once I had already built up a significantly large base with many lines of conduits spanning several dozen (loaded) chunks.
...

____
#### Affected Versions (Do *not* use "latest"):

- EnderIO: EnderIO-1.12.2-5.1.55
- EnderCore: EnderCore-1.12.2-0.5.73
- Minecraft: 1.12.2
- Forge: 14.23.5.2847
- SpongeForge? Yes
- Optifine? Yes
- Single Player and/or Server? Server

#### Your most recent log file where the issue was present:
https://pastebin.com/UyqE6J0h

#### Additional info:

I am playing the Omnifactory modpack. I asked about this issue in the discord a while back but couldn't get any concrete info as this seems to not be an inherent issue with the modpack itself.

[Here is a picture of it with a single conduit](https://i.imgur.com/u9FdJ2Y.png), I've had it happen to several conduits all in a line as well, and with every conduit type in the modpack.

The server I'm playing on where this is happening has a scheduled restart every 6 hours. This sometimes doesn't happen for days, and sometimes it happens two consecutive restarts in a row.  It *seems* to be mostly random. I've been talking to the server owner trying to figure out what's causing it to try and fix it. [We found a mildly related issue that seemed to point to FTBUtils chunkloading](https://github.com/FTBTeam/FTB-Utilities/issues/360) but that seems like a weird thing to cause them to break this way, as opposed to just not working while remaining connected like I've seen in other issues while searching. If you need additional info about the server and what it's running I can try to get that information if the owner is comfortable with that.

 All of the posts I've found about this issue seem to be on a 1.10 version of the mod which is weird since I'm having the issue in a 1.12 version so it still exists in some form, thus I would have expected to see at least one other post about it on a version after 1.10.
